### DISCUSSION TITLE
Allow custom paths without slug segment to store images

### DISCUSSION THREAD

**Discussion by NoelOConnell** [Thinkmill/keystatic#283] [Ideas]
I have an existing site with 300+ posts that I'm trying to migrate from Netlify CMS to Keystatic.

In the existing app all blog post images are stored like this `public/assets/posts/[image name].ext`
I have setup my `image` field like this:
```js
image: fields.image({
  label: "Cover image",
  directory: "public/assets/posts",
  description: "The cover image for the blog post",
}),
```

However, when I add a new post the image is saved in `public/assets/posts/[post-slug]/[image name].ext`
Is there anyway to stop the folder matching the post slug from being created?

**Comment by simonswiss** [comment]
Hey!

As of now, this is not possible since Keystatic depends on that slug structure/hierarchy to work out what image belongs to what entry when reading from the file system.

However, this is something that could _potentially_ be implemented in the future.

For right now though, you'd need to manually (or via a script) update the path to all those images to mirror the post slug.

Converted this to a discussion so we can continue the conversation here 👍

**Reply by NoelOConnell** [reply]
Thanks for the response.

Having the image path require the slug structure makes it awkward to migrate an existing site. It would be great if this limitation wasn't required.
Is there plans to have a "media library" page in the future? Would this be more difficult if all the images are nested in directories with slugs?

**Reply by simonswiss** [reply]
No concrete plans, but I've heard discussions of a media library internally. Nothing that is currently "in flight" though, so that won't be a thing for a little while at least.

I agree that there are limitations to having the path bound to the slug like that — that's something we plan to improve.
